tests/base/src/test/java/org/wildfly/security/auth/client/CredentialStoreSaslAuthenticationTest.java

```diff
@@ -0,0 +1,111 @@
+import org.wildfly.security.provider.util.ProviderFactory;
+import org.wildfly.security.sasl.plain.PlainSaslServerFactory;
+import org.wildfly.security.sasl.plain.WildFlyElytronSaslPlainProvider;
+import org.wildfly.security.sasl.test.SaslServerBuilder;
+import okhttp3.mockwebserver.MockWebServer;
+
+/**
+ * Tests a successful SASL authentication with a credential store reference in xml configuration
+ *
+ * @author <a href="mailto:szaldana@redhat.com">Sonia Zaldana</a>
+ */
+
+public class CredentialStoreSaslAuthenticationTest {
+
+    private static final String PLAIN = "PLAIN";
+    private static final String USERNAME = "Guest";
+    private static final String PASSWORD = "gpwd";
+    private static final String CREDENTIAL_CONFIG_FILE = "wildfly-credential-sasl-config.xml";
+    private static final MockWebServer server = new MockWebServer();
+    private static String BASE_STORE_DIRECTORY = "target/ks-cred-stores";
+    private static final Supplier<Provider[]> ELYTRON_PROVIDER_SUPPLIER = ProviderFactory.getElytronProviderSupplier(ElytronXmlParser.class.getClassLoader());
+
+    private static final Provider[] providers = new Provider[] {
+            WildFlyElytronSaslPlainProvider.getInstance(),
+            WildFlyElytronPasswordProvider.getInstance()
+    };
+
+    @BeforeClass
+    public static void setUp() throws Exception {
+        server.start(50831);
+        System.setProperty("wildfly.config.url", CredentialStoreSaslAuthenticationTest.class.getResource(CREDENTIAL_CONFIG_FILE).toExternalForm());
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        server.shutdown();
+        cleanCredentialStore();
+    }
+
+    private static void cleanCredentialStore() {
+        File dir = new File(BASE_STORE_DIRECTORY);
+        dir.mkdirs();
+        File file = new File(BASE_STORE_DIRECTORY + "/mycredstore.cs");
+        file.delete();
+    }
+
+    @Test
+    public void testSuccessfulSaslAuthenticationWithCredentialStore() throws Exception {
+
+            // Enable Elytron providers manually to configure credential store
+            for (Provider provider : providers) {
+                Security.insertProviderAt(provider, 1);
+            }
+
+            cleanCredentialStore();
+
+            CredentialStoreBuilder.get().setKeyStoreFile(BASE_STORE_DIRECTORY + "/mycredstore.cs")
+                    .setKeyStoreType("JCEKS")
+                    .setKeyStorePassword("StorePassword")
+                    .addPassword(USERNAME, PASSWORD)
+                    .build();
+
+            // disable Elytron providers before client and server exchange messages
+            for (Provider provider : providers) {
+                Security.removeProvider(provider.getName());
+            }
+
+            SaslServer server = new SaslServerBuilder(PlainSaslServerFactory.class, PLAIN)
+                    .setProviderSupplier(ELYTRON_PROVIDER_SUPPLIER)
```

Comment:
I think you should be able to use just the two required providers here (i.e., `() -> providers`) and then the unused `ELYTRON_PROVIDER_SUPPLIER` can be removed.

Reply:
fixed!